source/src/WebSocketServer.cc

```diff
@@ -0,0 +1,364 @@
+      WsServiceBase(s, n, RPC_TYPE) {
+      
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    WsRequestHandler::signal_type &WsRequestHandler::onRequest() {
+      return m_signal;
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    //-------------------------------------------------------------------------------------------------
+    
+    WsServer::WsServer() :
+      m_server() {
+      
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    WsServer::~WsServer() {
+      
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    void WsServer::setPort(int port) {
+      m_port = port;
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    WsService *WsServer::createService(const std::string &name) {
+      auto findIter = m_serviceMap.find(name);
+      if(findIter != m_serviceMap.end()) {
+        dqm_error("Couldn't create service '{0}' twice", name);
+        return nullptr; 
+      }
+      WsService *service = new WsService(this, name);
+      m_serviceMap.insert(ServiceMap::value_type(name, service));
+      m_serviceConnections[name] = connection_set();
+      return service;
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    WsCommandHandler *WsServer::createCommandHandler(const std::string &name) {
+      auto findIter = m_serviceMap.find(name);
+      if(findIter != m_serviceMap.end()) {
+        dqm_error("Couldn't create service '{0}' twice", name);
+        return nullptr; 
+      }
+      WsCommandHandler *service = new WsCommandHandler(this, name);
+      m_serviceMap.insert(ServiceMap::value_type(name, service));
+      return service;
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    WsRequestHandler *WsServer::createRequestHandler(const std::string &name) {
+      auto findIter = m_serviceMap.find(name);
+      if(findIter != m_serviceMap.end()) {
+        dqm_error("Couldn't create service '{0}' twice", name);
+        return nullptr; 
+      }
+      WsRequestHandler *service = new WsRequestHandler(this, name);
+      m_serviceMap.insert(ServiceMap::value_type(name, service));
+      return service;
+    }
+    
+    //-------------------------------------------------------------------------------------------------
+    
+    WsService *WsServer::findService(const std::string &name) {
+      auto findIter = m_serviceMap.find(name);
+      if(findIter == m_serviceMap.end()) {
+        return nullptr;
+      }
+      if(findIter->second->type() != SERVICE_TYPE) {
```

Comment:
You could still send a message saying you found something with the right name but the wrong Type.
It means there must be a problem, either you should not be looking for a service or you are looking for the wrong name.